src/OrchardCore/OrchardCore.ResourceManagement.Abstractions/ResourceDefinition.cs

```diff
@@ -139,7 +139,13 @@ public ResourceDefinition SetCdn(string cdnUrl, string cdnUrlDebug, bool? cdnSup
         /// <param name="version">The version to set, in the form of <code>major.minor[.build[.revision]]</code></param>
         public ResourceDefinition SetVersion(string version)
         {
+            if (!System.Version.TryParse(version, out _))
+            {
+                throw new FormatException();
```

Comment:
We can add a message here if we agreed to inject `IStringLocalizer`

Reply:
I don't think we usually tanslate exception messages. But I think it's ok to emit a meaningful message instead with `major.minor[.build[.revision]]`